### PR TITLE
Improve VS Code build current project support

### PR DESCRIPTION
**Pull request by jaredpar**
This makes the following changes to our ability to build the current
project from VSCode (essentially implementing the rough equivalent of
build current project in VS).

- Ability to choose `msbuild` in addition to `dotnet msbuild`. The full
framework `msbuild` is still necessary for building parts of the IDE
code base.
- Support for VB projects
- Fixes the execution of the task on Windows. The hard dependency on
`/.dotnet` existing doesn't hold on Windows. Falling back to normal
powershell in that environment

### CODE REVIEW

**Comment by jaredpar** [comment]
@333fred @RikkiGibson PTAL. I'm fairly confident I maintained the non-windows behavior for these tasks but would appreciate if you all could validate this. I have validated these changes work correctly on windows for both the `dotnet msbuild` and `msbuild` cases.

File: eng/build-utils.ps1 [hunk 1]
```diff
@@ -247,7 +247,7 @@ function Get-PackageDir([string]$name, [string]$version = "") {
   return $p
 }
 
-function Run-MSBuild([string]$projectFilePath, [string]$buildArgs = "", [string]$logFileName = "", [switch]$parallel = $true, [switch]$summary = $true, [switch]$warnAsError = $true, [string]$configuration = $script:configuration) {
+function Run-MSBuild([string]$projectFilePath, [string]$buildArgs = "", [string]$logFileName = "", [switch]$parallel = $true, [switch]$summary = $true, [switch]$warnAsError = $true, [string]$configuration = $script:configuration, [switch]$skipAnalyzers = $false) {
```
Comment (jaredpar) [inline eng/build-utils.ps1:250]: This function was relying on `$skipAnalyzers` being implicitly defined. That seemed dangerous so I had it be passed as an explicit argument and fixed up the callers to use this.

File: scripts/vscode-build.ps1 [hunk 2]
```diff
@@ -1,5 +1,6 @@
 param (
-  [Parameter(Mandatory = $true)][string]$filePath
+  [Parameter(Mandatory = $true)][string]$filePath,
+  [string]$msbuildEngine = "vs"
```
Comment (jaredpar) [inline scripts/vscode-build.ps1:3]: This parameter effectively acts like a global variable here. The implementation of `InitializeBuildTool` below relies on reading this variable to pick the appropriate build engine. 

This pattern is how our core `build.ps1` script works and I'm essentially replicating it here. That should make this script more resilient over time as it should evolve in sync with our core build logic.

File: eng/build-utils.ps1 [hunk 3]
```diff
@@ -181,7 +181,7 @@ function Get-ProjectFile([object]$fileInfo) {
   try {
     while ($true) {
       # search up from the current file for a folder containing a csproj
```
Comment (RikkiGibson) [inline eng/build-utils.ps1:183]:
```suggestion
      # search up from the current file for a folder containing a project file
```

File: scripts/vscode-build.ps1 [hunk 4]
```diff
@@ -10,17 +11,14 @@ $ErrorActionPreference = "Stop"
 $fileInfo = Get-ItemProperty $filePath
 $projectFileInfo = Get-ProjectFile $fileInfo
 if ($projectFileInfo) {
-  $dotnetPath = Resolve-Path (Ensure-DotNetSdk) -Relative
-  $projectDir = Resolve-Path $projectFileInfo.Directory -Relative
-
-  $invocation = "$dotnetPath msbuild $projectDir -p:UseRoslynAnalyzers=false -p:GenerateFullPaths=true"
-  Write-Output "> $invocation"
-  Invoke-Expression $invocation
+  $buildTool = InitializeBuildTool
```
Comment (333fred) [inline scripts/vscode-build.ps1:14]: Remember when I approved this? Yeah, that was a mistake.

File: scripts/vscode-build.ps1 (hunk 2 above)
Comment (333fred) [reply scripts/vscode-build.ps1:3]: And `InitializeBuildTool` is in arcade and hard coded to dotnet.exe :(

**Pull request merged**